components/elm/src/main/ncdio_pio.F90.in

```diff
@@ -2157,7 +2180,20 @@ end subroutine ncd_io_{DIMS}d_{TYPE}_glob
 
     allocate(dids(ndims))
     status = pio_inq_vardimid(ncid, vardesc, dids)
+
+    !find out if "time" is one of the dimnames
     do i = 1,ndims
+       status = pio_inq_dimname(ncid,dids(i),dimname)
+       dimname=shr_string_toLower(dimname)
+       if ('time' == trim(dimname)) then
+          ndims_iod = ndims - 1
+          if(present(time_dim_exists))time_dim_exists = .true. !set it to true if "time" dimension exists
+       else
+          ndims_iod = ndims
+       end if
+    enddo
+
+    do i = 1,ndims_iod
```

Comment:
@tktesfa2014 : Also please confirm that lspft does not need any special handling in scam_field_offsets() (With the pft dim only the data corresponding to the closest lat/lon are being read, lspft does not have similar code to read specific data)

Reply:
@jayeshkrishna as far as I understand pft does not need any special handling in scam_field_offsets().

Reply:
Thanks @singhbalwinder, @jayeshkrishna and @rljacob for your help on this branch!